model/project_parser_test.go

```diff
@@ -782,3 +783,239 @@ tasks:
 	assert.Equal("execTask2", proj.BuildVariants[0].DisplayTasks[1].ExecutionTasks[0])
 	assert.Equal("execTask4", proj.BuildVariants[0].DisplayTasks[1].ExecutionTasks[1])
 }
+
+func TestTaskGroupValidation(t *testing.T) {
```

Comment:
Add a null case, i.e., something that includes no task groups. I think it's unlikely that this case will error, but I think it's worth verifying that validate does what you expect for the no task group case.

Reply:
There are other tests in this package that make sure project yml parses without task groups, so I don't think it's necessary to add one here

Reply:
sgtm